samples/nrf_desktop/src/hw_interface/motion_optical.c

```diff
@@ -699,6 +733,15 @@ static void optical_thread_fn(void)
 			__ASSERT_NO_MSG(false);
 			break;
 		}
+
+		if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE)) {
+			static u16_t current_cpi;
```

Comment:
initial cpi is equal to `CONFIG_DESKTOP_OPTICAL_DEFAULT_CPI`

Reply:
for this reason I suggest to pass the cpi value to firmware load (and move this variable to the beginning of this function i.e.
```
func()
{
   static unsigned cpi = DEFAULT;
   ...
   firmware_load(cpi);
   ...
   if (new_cpi != cpi) {
      cpi = new_cpi;
      set_cpi(cpi);
   }
   ...
}
```